### ISSUE TITLE
Experiment with other ML algorithms

### ISSUE THREAD

**Issue by yujinjcho** [yujinjcho/movie_recommendations#11]
try random forrest, neural networks, svm and assess performance/time. others?

**Comment by yujinjcho** [comment]
@achiu8

Made a [branch](https://github.com/yujinjcho/movie_recommendations/tree/add-other-models) with some other models . I copied [this](http://scikit-learn.org/stable/auto_examples/classification/plot_classifier_comparison.html). Do you want to help me assess them?

Maybe do a little research on each and experiment with:
- speed
- accuracy
- easiest to tune

then just pick one or two we can use going forward

**Comment by achiu8** [comment]
Sure, do you want to split them up or each look into all of them?

**Comment by achiu8** [comment]
@yujinjcho Just realized you might not have gotten a notification because I didn't mention you..

**Comment by yujinjcho** [comment]
Oh sorry I saw. I don't mind both doing all of them

**Comment by yujinjcho** [comment]
Added [gridsearchcv](http://scikit-learn.org/stable/modules/grid_search.html#grid-search) and reorganized how data was getting split up

Might make tuning multiple params easier.
https://github.com/yujinjcho/movie_recommendations/tree/tune-with-gridcv

**Comment by yujinjcho** [comment]
Experimenting with automatic model selection / tuning with [auto-sklearn](https://automl.github.io/auto-sklearn/stable/) on this [branch](https://github.com/yujinjcho/movie_recommendations/tree/add-auto-tuner). Unfortunately super slow. maybe good for discovering rough upside potential?

```
Accuracy score 0.822972972973
Completed train - Minutes: 303, Seconds: 23
```

Need to run with python3

**Comment by yujinjcho** [comment]
For comparative purposes, neural net example using gridcv
```
parameters = {'alpha': [10, 30, 100], 'max_iter': [100, 300, 1000, 3000], 'hidden_layer_sizes': [(3,), (10,), (3, 3), (5, 5)]}
```
```
Finished tuning params - Minutes: 25, Seconds: 7
{'alpha': 30, 'max_iter': 300, 'hidden_layer_sizes': (5, 5)}
Stats:
----------
accuracy:  0.783783783784
f1: 0.832635983264
precision:  0.775828460039
recall:  0.89841986456
----------

```

**Comment by yujinjcho** [comment]
maybe we just use sklearn.linear_model.LogisticRegression
```
parameters = {'C': [0.01, 0.03, 0.1, 0.3, 1.0, 3.0, 10.0, 30.0, 100.0, 300.0, 1000.0]}
```

```
Finished tuning params - Minutes: 0, Seconds: 9
{'C': 0.01}
Predicting results
Finished Predictions
 - Minutes: 0, Seconds: 0
Stats:
----------
accuracy:  0.82027027027
f1: 0.851396648045
precision:  0.842920353982
recall:  0.860045146727
----------

```

**Comment by achiu8** [comment]
Wait so auto-sklearn took 5 hours?? And accuracy of last one is just barely worse.. is that just plain logistic regression like what we had originally?

**Comment by yujinjcho** [comment]
ya pretty much except built in sklearn one
